### PR TITLE
Fixes/additions to support state-sync again

### PR DESCRIPTION
**Pull request by darioush**

### CODE REVIEW

File: scripts/Dockerfile.build.ubuntu [hunk 1]
```diff
@@ -1,9 +1,11 @@
-# https://gallery.ecr.aws/ubuntu/ubuntu
-FROM public.ecr.aws/ubuntu/ubuntu:latest
+# https://gallery.ecr.aws/lts/ubuntu
+FROM public.ecr.aws/lts/ubuntu:latest
```
Comment (darioush) [inline scripts/Dockerfile.build.ubuntu:2]: this matches libc on the version of ubuntu we start in aws

File: scripts/Dockerfile.build.ubuntu [hunk 2]
```diff
@@ -1,9 +1,11 @@
...
 
 # "build-essential" for linker "cc"
 # "pkg-config" for utility to find OpenSSL
 # "libssl-dev" for development packages of openssl
+ARG DEBIAN_FRONTEND=noninteractive
+
```
Comment (darioush) [inline scripts/Dockerfile.build.ubuntu:8]: this avoids a silly prompt asking to input the timezone on package install (lol....)

File: src/utils/prometheus.rs [hunk 3]
```diff
@@ -678,11 +678,20 @@ fn test_parse_avalanchego_metrics() {
...
 pub fn match_metric<'a, F>(samples: &'a [Metric], f: F) -> &'a Metric
 where
     for<'r> F: FnMut(&'r &'a Metric) -> bool,
 {
-    samples.iter().find(f).as_ref().unwrap()
+    let metric = samples.iter().find(f);
+    if metric.is_some() {
+        return metric.unwrap();
+    }
+    return &NOT_FOUND_METRIC;
```
Comment (darioush) [inline src/utils/prometheus.rs:694]: wasn't sure what to do here but this would sometimes panic so I got creative.

Comment (gyuho) [reply src/utils/prometheus.rs:698]: ah, thanks for the fix!

**Review by gyuho — APPROVED** [review]

**Pull request merged**